### PR TITLE
dropout function always active

### LINKED ISSUE #1084: dropout function is implemented as just an identity
The forward pass of the `Dropout` layer, which gets called when you do something like`d = Dropout(p); d(x)` is defined as just an identity function [here](https://github.com/FluxML/Flux.jl/blob/df3f904f7c34f095562693b6a9ca67047319dea0/src/layers/normalise.jl#L22).

Is that a bug? Or am I going crazy?

### PR DESCRIPTION
**Pull request by CarloLucibello**
Fix #1084 according to the suggestions of @AzamatB and @jondeuce

### PR Checklist

- [x] Tests are added
- [x] Entry in NEWS.md
- [ ] Documentation, if applicable
- [ ] Final review from `@MikeInnes` or `@dhairyagandhi96` (for API changes).

### CODE REVIEW

**Comment by DhairyaLGandhi** [comment]
I don't think we should do it this way, dropout is usually omitted at test time, this way it will make it difficult to turn it off when most cases all for it to be off

**Comment by CarloLucibello** [comment]
> I don't think we should do it this way, dropout is usually omitted at test time, this way it will make it difficult to turn it off when most cases all for it to be off

this is what the `Dropout` type is for. In fact, in 99% cases you should always use the `Dropout` layer and never the `dropout` function.  With this PR the implementations are now consistent with the docstrings (they weren't before). I can add some additional warning in the docstring

**Comment by DhairyaLGandhi** [comment]
In that case better to just remove the starting `_` in `dropout_kernel` and document using that if anyone wants to dropout all the time. Fixing a docstring is a good idea though. Thanks!

**Comment by CarloLucibello** [comment]
to avoid silent change of behavior, I introduced the mandatory keyword argument `active`, that we can possibly deprecate in the future

File: NEWS.md [hunk 1]
```diff
@@ -17,6 +17,7 @@
 * Testing suite improvements now test for gradients of all layers along with GPU support.
 * Functors have now moved to [Functors.jl](https://github.com/FluxML/Flux.jl/pull/1174) to allow for their use outside of Flux.
 * Added [helper functions](https://github.com/FluxML/Flux.jl/pull/873) `Flux.convfilter` and `Flux.depthwiseconvfilter` to construct weight arrays for convolutions outside of layer constructors so as to not have to depend on the default layers for custom implementations.
+* `dropout` function is now [always active](https://github.com/FluxML/Flux.jl/pull/1263)
```
Comment (oxinabox) [inline NEWS.md:20]: mention the `Dropout` layer here ?

**Comment by CarloLucibello** [comment]
@DhairyaLGandhi merge?

File: NEWS.md [hunk 2]
```diff
@@ -17,6 +17,8 @@
 * Testing suite improvements now test for gradients of all layers along with GPU support.
 * Functors have now moved to [Functors.jl](https://github.com/FluxML/Flux.jl/pull/1174) to allow for their use outside of Flux.
 * Added [helper functions](https://github.com/FluxML/Flux.jl/pull/873) `Flux.convfilter` and `Flux.depthwiseconvfilter` to construct weight arrays for convolutions outside of layer constructors so as to not have to depend on the default layers for custom implementations.
+* `dropout` function now has a mandatory [active](https://github.com/FluxML/Flux.jl/pull/1263)
+keyword argument. The `Dropout` struct is the recommended choice for common usage.
```
Comment (DhairyaLGandhi) [inline NEWS.md:21]: Maybe say the behaviour of the layer has not changed.

**Comment by CarloLucibello** [comment]
@DhairyaLGandhi done

**Comment by DhairyaLGandhi** [comment]
bors r+

In general, we should try to remove the dependency on the `active` field in other structs as well, and use functional approaches instead (like we used to do earlier) and remove `trainmode!` but that's breaking and the matter of a separate discussion.

**Pull request merged**